README.md

```diff
@@ -292,38 +292,48 @@ GraphQLObjectType person = newObject()
  
 #### Data fetching
 
-The actual data comes from `DataFetcher` objects.
- 
-Every field definition has a `DataFetcher`. When no one is configured, a 
-[PropertyDataFetcher](src/main/java/graphql/schema/PropertyDataFetcher.java) is used.
+Actual field data comes from [`DataFetcher`](src/main/java/graphql/schema/DataFetcher.java) objects that 
+are specified for every field definition. graphql-java comes with four built-in data fetcher 
+implemenations. When a field does not specify a data fetcher, 
+the [PropertyDataFetcher](src/main/java/graphql/schema/PropertyDataFetcher.java) is used.
 
-`PropertyDataFetcher` fetches data from `Map` and Java Beans. So when the field name matches the Map key or
-the property name of the source Object, no `DataFetcher` is needed. 
+The built-in implemenations are:
 
+* [`StaticDataFetcher`](src/main/java/graphql/schema/StaticDataFetcher.java): Resolves a field with pre-defined, static data.
+* [`MapDataFetcher`](src/main/java/graphql/schema/MapDataFetcher.java): Dereferences a key from the source object.
+  If the source object is not a map, or if the key is not set on the map, returns `null`.
+* [`FieldDataFetcher`](src/main/java/graphql/schema/FieldDataFetcher.java): Gets a public field from the source object. 
+* [`PropertyDataFetcher`](src/main/java/graphql/schema/PropertyDataFetcher.java): Gets a property from the source object.
+  If the field type is boolean, attempts using a `is`- or `get`-prefixed getter method (in that order). Otherwise, a `get`-prefixed
+  getter is used.
 
+When the built-in data fetchers are insufficient, a custom `DataFetcher` implementations is easy to define:
 
-Example of configuring a custom `DataFetcher`:
-```java
+```
+newFieldDefinition()
+    .name("fieldName")
+    .type(GraphQLString)
+    .dataFetcher(new DataFetcher() {
+        @Override
+        Object get(DataFetchingEnvironment env) {
+            // env.getSource() is the value of the surrounding object. 
+            Object source = env.getSource() 
```

Comment:
Kind of wish this was a better example anyway. Also this section of the docs should document `DataFetchingEnvironment`, right?